easybuild/framework/easyblock.py

```diff
@@ -2380,8 +2380,13 @@ def handle_iterate_opts(self):
             # only needs to be done during first iteration, since after that the options won't be lists anymore
             if self.iter_idx == 0:
                 # keep track of list, supply first element as first option to handle
+                iter_cnt = self.det_iter_cnt()
                 for opt in self.cfg.iterate_options:
-                    self.iter_opts[opt] = self.cfg[opt]  # copy
+                    if isinstance(self.cfg[opt], (list, tuple)):
+                        self.iter_opts[opt] = self.cfg[opt]  # copy
+                    elif iter_cnt > 1:
+                        # make iter_cnt copies for every opt since easyblocks can modify them
+                        self.iter_opts[opt] = [self.cfg[opt]] * iter_cnt
```

Comment:
Although this will help, it's not a perfect solution since for some easyconfig parameters (which have a list or tuple value but are not iterated over), we won't be doing a proper reset...

Can we come up with a more robust way or resetting *all* easyconfig parameters before we start a new iteration?

Reply:
it's only a few easyconfig parameters that are affected by this logic:
```
ITERATE_OPTIONS = ['builddependencies',
                   'preconfigopts', 'configopts', 'prebuildopts', 'buildopts', 'preinstallopts', 'installopts']
```
which are normally always strings right?

The exception being builddependencies; I will have to special case that one here I think.

Reply:
oh I may have misunderstood the *all* easyconfig parameters, it's more a wish list.

IMHO that would be a much bigger change, not for a bugfix release. You could also argue that a reset is only really needed for all easyconfig parameters that are affected by iteration; there really is no point doing it for checksums and sources, though one could reasonably make things like `pretestopts` and `testopts` iterable *and* saved (they aren't now).

I'd then much rather than go for a way of making most easyconfig parameters read-only, i.e. `__setitem__` can complain when an easyblock changes something it shouldn't.